addons/io_scene_gltf2/blender/imp/gltf2_blender_node.py

```diff
@@ -119,43 +133,50 @@ def create_bone(gltf, vnode_id):
         bpy.ops.object.mode_set(mode="EDIT")
-        editbone = armature.edit_bones.new(vnode.name)
-        vnode.blender_bone_name = editbone.name
 
-        # Set extras (if came from a glTF node)
-        if isinstance(vnode_id, int):
-            pynode = gltf.data.nodes[vnode_id]
-            set_extras(editbone, pynode.extras)
+        for id in bone_ids:
+            vnode = gltf.vnodes[id]
+            editbone = armature.edit_bones.new(vnode.name)
+            vnode.blender_bone_name = editbone.name
+            editbone.use_connect = False  # TODO?
+
+            # Give the position of the bone in armature space
+            arma_mat = vnode.bone_arma_mat
+            if bpy.app.version < (2, 80, 0):
+                editbone.head = arma_mat * Vector((0, 0, 0))
+                editbone.tail = arma_mat * Vector((0, 1, 0))
+                editbone.align_roll(arma_mat * Vector((0, 0, 1)) - editbone.head)
+            else:
+                editbone.head = arma_mat @ Vector((0, 0, 0))
+                editbone.tail = arma_mat @ Vector((0, 1, 0))
+                editbone.align_roll(arma_mat @ Vector((0, 0, 1)) - editbone.head)
 
-        # TODO
-        editbone.use_connect = False
+            if isinstance(id, int):
+                pynode = gltf.data.nodes[id]
+                set_extras(editbone, pynode.extras)
 
-        # Give the position of the bone in armature space
-        arma_mat = vnode.bone_arma_mat
-        if bpy.app.version < (2, 80, 0):
-            editbone.head = arma_mat * Vector((0, 0, 0))
-            editbone.tail = arma_mat * Vector((0, 1, 0))
-            editbone.align_roll(arma_mat * Vector((0, 0, 1)) - editbone.head)
-        else:
-            editbone.head = arma_mat @ Vector((0, 0, 0))
-            editbone.tail = arma_mat @ Vector((0, 1, 0))
-            editbone.align_roll(arma_mat @ Vector((0, 0, 1)) - editbone.head)
+        # Set all bone parents
+        for id in bone_ids:
+            vnode = gltf.vnodes[id]
+            parent_vnode = gltf.vnodes[vnode.parent]
+            if parent_vnode.type == VNode.Bone:
+                editbone = armature.edit_bones[vnode.blender_bone_name]
+                parent_editbone = armature.edit_bones[parent_vnode.blender_bone_name]
+                editbone.parent = parent_editbone
 
-        # Set parent
-        parent_vnode = gltf.vnodes[vnode.parent]
-        if parent_vnode.type == VNode.Bone:
-            editbone.parent = armature.edit_bones[parent_vnode.blender_bone_name]
+        # Switch back to object mode and do pose bones
 
         bpy.ops.object.mode_set(mode="OBJECT")
         pose_bone = blender_arma.pose.bones[vnode.blender_bone_name]
 
-        # Put scale on the pose bone (can't go on the edit bone)
-        _, _, s = vnode.trs
-        pose_bone.scale = s
+        for id in bone_ids:
+            # Put scale on pose bone (edit bones have no scale)
+            _, _, s = vnode.trs
```

Comment:
Seems you forgot to update vnode here.
vnode = gltf.vnodes[id]

Reply:
You're right. And the pose_bone too. Thanks!